gsms.py

```diff
@@ -53,15 +53,83 @@
 
 # standard imports
 import argparse
+import ctypes
+from ctypes import wintypes
 import json
 import os
+import re
 import shutil
 import time
+import uuid
 
 # lib imports
 import pylnk3
 
 
+# Code from here and modified to work in this project
+# https://gist.github.com/mkropat/7550097
+class GUID(ctypes.Structure):
+    """
+    Class to build a GUID compliant object for use in WIndows libraries
+
+    Parameters
+    -------
+    uuid: string
+
+    Returns
+    -------
+    GUID object
+
+    Examples
+    --------
+    >>> GUID(uuid.UUID("{B4BFCC3A-DB2C-424C-B029-7FE99A87C641}"))
+    ...
+    """
+    _fields_ = [
+        ("Data1", wintypes.DWORD),
+        ("Data2", wintypes.WORD),
+        ("Data3", wintypes.WORD),
+        ("Data4", wintypes.BYTE * 8)
+    ]
+
+    def __init__(self, uuid_):
```

Comment:
add type hint and return type (`None` in this case)

```py
def __init__(self, uuid_: str) -> None:
```

Reply:
Will add return types to functions added in the PR

Reply:
also added the parameter types. somehow didnt see that when reading the suggested code